src/components/Pokemon.vue

```diff
@@ -0,0 +1,31 @@
+<template>
+  <div>
+    <router-link to="/"><h1>Home</h1></router-link>
+    <card v-if="pokemon.id" :pokemon="pokemon"/>
+  </div>
+</template>
+
+<script>
+    import Card from './Card';
+    import PokemonService from './pokemonService'
+
+    export default {
+        name: "pokemon",
+        components: {Card},
+        data() {
+          return {
+            pokemon: {}
+          }
+        },
+        mounted() {
+          PokemonService.getPokemon(this.$route.params.id).then(pokemon => {
```

Comment:
To avoid a dependency between the router and component, you can add a property `props: true` in your route definition. With this props activated, vue-router will inject $route.params as property of the component.
I think it's a new step in your logic, so you can add this refacto in a new commit